src/code/z_play.c

```diff
@@ -1,81 +1,1608 @@
+
+    if (R_PICTOGRAPH_PHOTO_STATE == PICTOGRAPH_PHOTO_STATE_TAKE) {
+        R_PICTOGRAPH_PHOTO_STATE = PICTOGRAPH_PHOTO_STATE_FINISH;
+        MsgEvent_SendNullTask();
+        Play_TakePictographPhoto(&this->pauseBgPreRender);
+        R_PICTOGRAPH_PHOTO_STATE = PICTOGRAPH_PHOTO_STATE_OFF;
+    }
+    Actor_SetMovementScale(this->state.framerateDivisor);
+
+    if (FrameAdvance_Update(&this->frameAdvCtx, &pad58[1])) {
+        if ((this->transitionMode == TRANS_MODE_OFF) && (this->transitionTrigger != TRANS_TRIGGER_OFF)) {
+            this->transitionMode = TRANS_MODE_SETUP;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/D_801DFA18.s")
+        if (gTrnsnUnkState != 0) {
+            switch (gTrnsnUnkState) {
+                case 2:
+                    if (TransitionUnk_Init(&sTrnsnUnk, 10, 7) == NULL) {
+                        gTrnsnUnkState = 0;
+                    } else {
+                        D_801F6D0C = gZBufferPtr;
+                        gTrnsnUnkState = 3;
+                        Game_SetFramerateDivisor(&this->state, 1);
+                    }
+                    break;
+                case 3:
+                    func_801647AC(&sTrnsnUnk);
+                    break;
+            }
+        }
+        Play_UpdateTransition(this);
+        if (gTrnsnUnkState != 3) {
+            if ((gSaveContext.gameMode == 0) &&
+                (((this->msgCtx.msgMode == 0)) ||
+                 ((this->msgCtx.currentTextId == 0xFF) && (this->msgCtx.msgMode == 0x42) &&
+                  (this->msgCtx.unk12020 == 0x41)) ||
+                 ((this->msgCtx.currentTextId >= 0x100) && (this->msgCtx.currentTextId <= 0x200))) &&
+                (this->gameOverCtx.state == GAMEOVER_INACTIVE)) {
+                KaleidoSetup_Update(this);
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_80166060.s")
+            sp5C = (this->pauseCtx.state != 0) || (this->pauseCtx.debugEditor != DEBUG_EDITOR_NONE);
+
+            AnimationContext_Reset(&this->animationCtx);
+            Object_UpdateBank(&this->objectCtx);
+
+            if ((sp5C == 0) && (IREG(72) == 0)) {
+                this->gameplayFrames++;
+                Rumble_SetUpdateEnabled(true);
+                if ((this->actorCtx.freezeFlashTimer != 0) && (this->actorCtx.freezeFlashTimer-- < 5)) {
+                    pad60 = this->actorCtx.freezeFlashTimer;
+                    if ((pad60 > 0) && ((pad60 % 2) != 0)) {
+                        this->envCtx.fillScreen = true;
+                        this->envCtx.screenFillColor[0] = this->envCtx.screenFillColor[1] =
+                            this->envCtx.screenFillColor[2] = 150;
+                        this->envCtx.screenFillColor[3] = 80;
+                    } else {
+                        this->envCtx.fillScreen = false;
+                    }
+                } else {
+                    Room_HandleLoadCallbacks(this, &this->roomCtx);
```

Comment:
this isnt a change from this PR, but I am curious on this choice of name if anyone knows. theres no callbacks occurring anywhere in this afaik, but may just be misunderstanding the intent

Reply:
Name came from eons ago, before `z_room.c` even existed it looks like, let alone decomped
https://github.com/zeldaret/mm/commit/9bc4990814d2e953df294bc57f7bca1918bd3bac

Reply:
I guess the rationale behind the name is that it doesn't really do anything unless the DMA is complete, so it's a "callback" with respect to DMA completion. But yeah there are better names for sure

Reply:
IIRC z_room looks very similar between the games, so it may have a direct OoT equivalent anyway.